### PR TITLE
Add Heroku staging app + pipeline

### PR DESCRIPTION
**Pull request by dchiquito**

### CODE REVIEW

**Comment by mgrauer** [comment]
Cool! Can you explain what this does, and how it relates to code branches/S3/database?

File: terraform/staging_pipeline.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  app      = "dandi-api-staging"
+  pipeline = heroku_pipeline.dandi_pipeline.id
+  stage    = "staging"
+}
+
+resource "heroku_pipeline_coupling" "production" {
+  app      = "dandi-api"
+  pipeline = heroku_pipeline.dandi_pipeline.id
+  stage    = "production"
+}
```
Comment (waxlamp) [inline terraform/staging_pipeline.tf:54]: These resources are the special sauce for dealing with pipelines as opposed to apps, is that right?

**Review by waxlamp — APPROVED** [review]

**Comment by dchiquito** [comment]
@mgrauer This is more of a first draft. After applying these changes, production will be completely unaffected, and there will be a staging app with some sensible default configuration. However, the staging app won't be hooked up to GitHub, so there will be no code deployed to it. That needs to be hooked up manually.

Right now, production is deploying automatically whenever `dandiapi/master` updates. I'm honestly not sure exactly what is possible with pipelines, but my plan is to make staging deploy directly from `master`, which can then be promoted to production manually from the heroku console.

Once the pipeline is in place and some experimentation is done, we can make better informed decisions about what the development flow would be. Perhaps a `dev` branch that deploys to `staging`, so `master` deploys directly to production, and we can add some version release stuff to merge dev into master with a version bump or whatever.

File: terraform/staging_pipeline.tf (hunk 1 above)
Comment (dchiquito) [reply terraform/staging_pipeline.tf:54]: Correct, this is what sets up the pipeline + adds the apps to the pipeline.

**Pull request merged**